### ISSUE TITLE
dapr-operator CrashLoopBackOff when upgrade from 1.10.6 to 1.11.0

### ISSUE THREAD

**Issue by et2012** [dapr/dapr#6498]
## In what area(s)?

/area operator

## What version of Dapr?

from: 1.10.6
to: 1.11.0

## Expected Behavior

Upgrade successfully from 1.10.6 to 1.11.0

## Actual Behavior

dapr-operator CrashLoopBackOff

## Steps to Reproduce the Problem

dapr upgrade -k --runtime-version 1.11.0

```
time="2023-06-12T03:18:57.708814406Z" level=info msg="log level set to: info" instance=dapr-operator-745b6b4f5c-zndwh scope=dapr.operator type=log ver=1.11.0
time="2023-06-12T03:18:57.708933411Z" level=info msg="metrics server started on :9090/" instance=dapr-operator-745b6b4f5c-zndwh scope=dapr.metrics type=log ver=1.11.0
time="2023-06-12T03:18:57.708986875Z" level=info msg="starting Dapr Operator -- version 1.11.0 -- commit 58f50c943b893f2bc62941c4400d9b2293f7890e" instance=dapr-operator-745b6b4f5c-zndwh scope=dapr.operator type=log ver=1.11.0
time="2023-06-12T03:18:59.520543053Z" level=info msg="Dapr Watchdog is not enabled" instance=dapr-operator-745b6b4f5c-zndwh scope=dapr.operator type=log ver=1.11.0
time="2023-06-12T03:18:59.520686232Z" level=info msg="Dapr Operator is starting" instance=dapr-operator-745b6b4f5c-zndwh scope=dapr.operator type=log ver=1.11.0
time="2023-06-12T03:18:59.621272338Z" level=info msg="Healthz server is listening on :8080" instance=dapr-operator-745b6b4f5c-zndwh scope=dapr.operator type=log ver=1.11.0
I0612 03:18:59.621464       1 leaderelection.go:248] attempting to acquire leader lease dapr-system/operator.dapr.io...
time="2023-06-12T03:18:59.722642352Z" level=info msg="Getting TLS certificates" instance=dapr-operator-745b6b4f5c-zndwh scope=dapr.operator type=log ver=1.11.0
time="2023-06-12T03:18:59.722706328Z" level=info msg="TLS certificates loaded successfully" instance=dapr-operator-745b6b4f5c-zndwh scope=dapr.operator type=log ver=1.11.0
time="2023-06-12T03:18:59.722718739Z" level=info msg="Starting watch for certs on filesystem: /var/run/dapr/credentials" instance=dapr-operator-745b6b4f5c-zndwh scope=dapr.operator type=log ver=1.11.0
time="2023-06-12T03:18:59.722760768Z" level=info msg="Healthz server is shutting down" instance=dapr-operator-745b6b4f5c-zndwh scope=dapr.operator type=log ver=1.11.0
time="2023-06-12T03:18:59.725207165Z" level=info msg="The webhook CA secret was not found. Assuming conversion webhook caBundles are managed manually." instance=dapr-operator-745b6b4f5c-zndwh scope=dapr.operator type=log ver=1.11.0
time="2023-06-12T03:18:59.725221947Z" level=info msg="Starting api server" instance=dapr-operator-745b6b4f5c-zndwh scope=dapr.operator type=log ver=1.11.0
time="2023-06-12T03:18:59.725230361Z" level=info msg="starting gRPC server on port 6500" instance=dapr-operator-745b6b4f5c-zndwh scope=dapr.operator.api type=log ver=1.11.0
time="2023-06-12T03:18:59.726354498Z" level=fatal msg="error running operator: error running operator: unable to get http endpoint informer: no matches for kind \"HTTPEndpoint\" in version \"dapr.io/v1alpha1\"" instance=dapr-operator-745b6b4f5c-zndwh scope=dapr.operator type=log ver=1.11.0
```

**Comment by et2012** [comment]
I tried to list CRDs under `apiextensions.k8s.io` and found `httpendpoints.dapr.io` was missing,
after creating it by the code at `dapr/charts/dapr/crds/httpendpoints.yaml`, problem resolved.
this file was not applied somehow during upgrading.

**Comment by yaron2** [comment]
You need to be using the 1.11 version of the CLI in order to upgrade, it will be released in the next 24 hours.

**Comment by et2012** [comment]
I also encounted this error when I was upgrading `dapr` from `1.10.6` to `1.11.0` with CLI `1.10.0`.

```
Failed to upgrade Dapr: cannot patch "dapr-placement-server" with kind StatefulS
et: StatefulSet.apps "dapr-placement-server" is invalid: spec: Forbidden: update
s to statefulset spec for fields other than 'replicas', 'template', 'updateStrat
egy', 'persistentVolumeClaimRetentionPolicy' and 'minReadySeconds' are forbidden
```

after I mannually created the crd `httpendpoints.dapr.io` , it seems that `dapr-operator` works, and `dapr status -k` shows all components are running. but `helm list` indicates an error status of  `dapr` release because `dapr-placement-server` failed to patch.

I have updated CLI to `1.11.0` and run `dapr upgrade --runtime-version 1.11.0 -k`  again, but it still quits with above error.
but I didn't edit `yaml` of `dapr-placement-server` mannually.
is there a way to replace it forcelly? or do I need to upgrade with `helm install --replace` ?
